packages/ragbits-document-search/src/ragbits/document_search/retrieval/rerankers/llm_reranker.py

```
@@ -0,0 +1,131 @@
+        scoring_results = list(zip(flat_elements, scored_elements, strict=True))
+        scoring_results.sort(key=lambda x: x[1], reverse=True)
+        return [elem for elem, _ in scoring_results[: reranker_options.top_n]]
+
+    async def _score_elements(self, elements: Sequence[Element], query: str) -> Sequence[float]:
+        """
+        Scoring the elements according to their relevance to the query
+        Args:
+            elements: The sequence of elements to score.
+            query: The query to score elements with.
+
+        Returns:
+            A sequence of assigned scores.
+        """
+        logit_bias = self.get_yes_no_token_ids()
+        if logit_bias is not None:
+            self.llm_options.logit_bias = logit_bias
+
+        scored_elements = []
+        for doc in elements:
+            full_prompt = self.prompt_template.format(query=query, document=doc.text_representation)
+
+            prompt = SimplePrompt(content=full_prompt)
+
+            try:
+                res = await self.llm.generate_raw(prompt=prompt, options=self.llm_options)
+                answer = res["response"]
+                logprob = res["logprobs"][0]["logprob"]
+                prob = math.exp(logprob) if answer == "Yes" else 1 - math.exp(logprob)
+                scored_elements.append(prob)
+            except LLMStatusError as e:
+                raise NotImplementedError(
+                    f"Model {self.llm.model_name} doesn't support logprobs, "
+                    "which are crucial for this reranking method. Try to use other reranker."
+                ) from e
+
+        return scored_elements
+
+    def get_yes_no_token_ids(self) -> dict[int, int] | None:
+        """
+        Getting token ids for yes/no.
+
+        Returns:
+            logit_bias dict
+        """
+        tokens = [" Yes", " No"]
+        try:
+            model_name = litellm.get_llm_provider(self.llm.model_name)[0]
+        except Exception:
```

Please don't hide all exceptions. If there is a specific scenario that you want to handle in this way - catch the specific exception

---

changed